mlos_bench/mlos_bench/environments/composite_env.py

```diff
@@ -192,14 +193,46 @@ def run(self) -> Tuple[Status, Optional[dict]]:
             be in the `score` field.
         """
         _LOG.info("Run: %s", self._children)
-        (status, _) = result = super().run()
+        (status, metrics) = super().run()
         if not status.is_ready():
-            return result
+            return (status, metrics)
+
+        joint_metrics = {}
         for env_context in self._child_contexts:
             _LOG.debug("Child env. run: %s", env_context)
-            (status, _) = result = env_context.run()
-            _LOG.debug("Child env. run results: %s :: %s", env_context, result)
+            (status, metrics) = env_context.run()
+            _LOG.debug("Child env. run results: %s :: %s %s", env_context, status, metrics)
             if not status.is_good():
-                break
-        _LOG.info("Run completed: %s :: %s", self, result)
-        return result
+                _LOG.info("Run failed: %s :: %s", self, status)
+                return (status, None)
+            joint_metrics.update(metrics or {})
+
+        _LOG.info("Run completed: %s :: %s %s", self, status, joint_metrics)
+        return (status, joint_metrics)
+
+    def status(self) -> Tuple[Status, List[Tuple[datetime, str, Any]]]:
+        """
+        Check the status of the benchmark environment.
+
+        Returns
+        -------
+        (benchmark_status, telemetry) : (Status, list)
+            A pair of (benchmark status, telemetry) values.
+            `telemetry` is a list (maybe empty) of (timestamp, metric, value) triplets.
+        """
+        (status, telemetry) = super().status()
+        if not status.is_ready():
+            return (status, telemetry)
+
+        joint_telemetry = []
+        final_status = None
+        for env_context in self._child_contexts:
+            (status, telemetry) = env_context.status()
+            _LOG.debug("Child env. status: %s :: %s", env_context, status)
+            joint_telemetry.extend(telemetry)
+            if not status.is_good() and final_status is None:
+                final_status = status
```

Comment:
Wonder if there's a `max(final_status, status)` version of this that propogates the "worst" status seen, but I think this is good enough for now.

Reply:
I was thinking about it - but the meaning of "best" and "worst" status can vary across different contexts.. it is very likely I'll get back to this issue, though - when we implement the environment resetting properly